server/src/storage/compressed_data_buffer.rs

```diff
@@ -0,0 +1,169 @@
+        debug_assert!(
+            !self.compressed_segments.is_empty(),
+            "Cannot save CompressedDataBuffer with no data."
+        );
+
+        // Combine the compressed segments into a single RecordBatch.
+        let batch =
+            compute::concat_batches(&compressed_schema.0, &self.compressed_segments).unwrap();
+
+        // Create the folder structure if it does not already exist.
+        fs::create_dir_all(folder_path)?;
+
+        // Create a new file that includes the start timestamp of the first segment in batch, the
+        // end timestamp of the last segment in batch, the minium value stored in batch, and the
+        // maximum value stored in batch as the file name to efficiently pruning files that only
+        // contains data points with timestamps and values that is not relevant for a given query.
+        let file_name = storage::create_time_and_value_range_file_name(&batch);
+        let file_path = folder_path.join(file_name);
+        StorageEngine::write_batch_to_apache_parquet_file(batch, file_path.as_path())
+            .map_err(|error| IOError::new(Other, error.to_string()))?;
+
+        Ok(())
+    }
+
+    /// Return the size in bytes of `compressed_segments`.
+    fn get_size_of_compressed_segments(compressed_segments: &RecordBatch) -> usize {
+        let mut total_size: usize = 0;
+
+        // Compute the total number of bytes of memory used by the columns.
+        for column in compressed_segments.columns() {
+            total_size += column.get_array_memory_size()
+        }
+
+        total_size
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use tempfile::tempdir;
+
+    use crate::metadata::test_util as metadata_test_util;
+    use crate::storage::test_util;
+
+    #[test]
+    fn test_can_append_valid_compressed_segments() {
+        let mut time_series = CompressedDataBuffer::new();
+        time_series.append_compressed_segments(test_util::get_compressed_segments_record_batch());
+
+        assert_eq!(time_series.compressed_segments.len(), 1)
+    }
+
+    #[test]
+    fn test_compressed_time_series_size_updated_when_appending() {
+        let mut time_series = CompressedDataBuffer::new();
+        time_series.append_compressed_segments(test_util::get_compressed_segments_record_batch());
+
+        assert!(time_series.size_in_bytes > 0);
+    }
+
+    #[test]
+    fn test_can_save_compressed_segments_to_apache_parquet() {
+        let mut time_series = CompressedDataBuffer::new();
+        let segment = test_util::get_compressed_segments_record_batch();
+        time_series.append_compressed_segments(segment.clone());
+
+        let temp_dir = tempdir().unwrap();
+        time_series
+            .save_to_apache_parquet(
+                temp_dir.path(),
+                &metadata_test_util::get_compressed_schema(),
+            )
+            .unwrap();
+
+        // Data should be saved to a file with the first start time and last end time as the file
```

Comment:
Maybe update this comment to also mention the value range.

Reply:
Fixed.